soc/projects/science.md

```diff
@@ -0,0 +1,59 @@
+
+**Recommended Skills**: Background knowledge in quantum information and tensor networks and the ability of coding with Julia.
+**Expected Results**: Provide Yao.jl an extension package that defines different kinds of noise and provide Yao.jl with the ability to simulate certain kind of noisy circuits efficiently.
+**Mentors**: [Roger Luo](https://github.com/Roger-luo/)
+
+## Quantum Computation: Visualization of Quantum Circuits
+
+Although, there is already a pretty printing for quantum circuits in [Yao.jl](https://github.com/QuantumBFS/Yao.jl) as a tree, we are still lack of visualizing a quantum circuit defined as [block tree](https://quantumbfs.github.io/Yao.jl/latest/man/blocks/). And we have already have basic plotting utilities in Julia, like [Luxor.jl](https://github.com/JuliaGraphics/Luxor.jl). It would be great and more convenient to provide multiple theme for plotting a quantum circuit with Yao.jl to various formats.
+
+**Recommended Skills**: Basic knowledge in plotting in Julia with Luxor.jl.
+**Expected Results**: Provide Yao.jl an extension package that converts a block tree to an image, which contains not only the circuit but also mark of composite blocks, users should be able to change theme for paper or multi-media use.
+**Mentors**: [Roger Luo](https://github.com/Roger-luo/)
+**References**: some demo image can be found in Yao.jl's [doc](https://quantumbfs.github.io/Yao.jl/latest/tutorial/QFT/)
+
+
+## Quantum Computation: Tensor Network for Quantum Circuit Simulation
+
+A Tensor network is a constraction of tensors, it has wide applications in physics and machine learning. Especially, the similation scheme based on treewidth algorithm help us better simulate intermediate size quantum circuits with limited entanglement.
+
+> Simulating quantum computation by contracting tensor networks
+> Igor L. Markov, Yaoyun Shi
+> https://arxiv.org/abs/quant-ph/0511069
+
+To make this task interesting, we will create a UTF-8 based pictograph for tensor operations, making the contraction process more intuitive. And this task is very suited for Julia, since it supports not only [UTF-8 character input](https://docs.julialang.org/en/v0.6.0/manual/unicode-input/), but also [UTF-8 operators](https://github.com/JuliaLang/julia/blob/c200b4cdb9620b6df369ae3c735cf3af30b6a47f/src/julia-parser.scm). For the details, please read this [proposal](https://github.com/QuantumBFS/FunnyTN.jl/blob/master/docs/src/dev/proposal.md).
+
+**Recommended Skills:** Background knowledge in Julia language and linear algebra.
```

Comment:
Linear algebra to a student means the first linear algebra course, the first math course where they learn matrix multiplication. Is that sufficient?